api/queries_pr.go

```diff
@@ -799,6 +801,38 @@ func PullRequestReopen(client *Client, repo ghrepo.Interface, pr *PullRequest) e
 	return err
 }
 
+func PullRequestMerge(client *Client, repo ghrepo.Interface, pr *PullRequest) error {
+	return merge(client, repo, pr, githubv4.PullRequestMergeMethodMerge)
+}
+
+func PullRequestRebase(client *Client, repo ghrepo.Interface, pr *PullRequest) error {
+	return merge(client, repo, pr, githubv4.PullRequestMergeMethodRebase)
+}
+
+func PullRequestSquash(client *Client, repo ghrepo.Interface, pr *PullRequest) error {
```

Comment:
Instead of having a separate method for each of the merge options, could we pass a constant as an argument to a single method called e.g. `PullRequestMerge()`? We'd define 3 constants in the `api` package and then inside the method we would translate them to appropriate `githubv4` constants. This is the approach @vilmibm took [in `pr review`](https://github.com/cli/cli/blob/c40f064baa6210e52b29e6d10ec8e1783e1c8583/api/queries_pr.go#L618-L624) and I dig it!

Reply:
I originally tried something like this but ended up having lots of type conversion issues. I'll try this switch statement idea

Reply:
Ahhh... this worked great. Thanks for the example link.